test/unit/composition.test.js

```diff
@@ -0,0 +1,385 @@
+'use strict';
+
+const Support = require('../support');
+const { expect } = require('chai');
+const Sequelize = Support.Sequelize;
+const expectsql = Support.expectsql;
+
+const {
+  Slot,
+  Placeholder,
+  Composition,
+  CompositionGroup
+} = require('../../lib/dialects/abstract/query-generator/composition');
+
+const {
+  QueryProto,
+  SelectProto,
+  InsertProto,
+  UpdateProto
+} = require('../../lib/dialects/abstract/query-generator/query-proto');
+
+describe('Composition', () => {
+
+  it('validates input', () => {
+    expect(() => new Composition('foo ', new Slot(new Date(), { type: Sequelize.DATE }), ' bar')).to.not.throw();
+    expect(() => new Composition('bar', new Placeholder())).to.not.throw();
+    expect(() => new Composition('numbers not allowed', 8008)).to.throw('Invalid query item').with.property('item', 8008);
+    const date = new Date(Date.UTC(2000, 0, 1));
+    expect(() => new Composition('Only allows strings, slots and placholders', date))
+      .to.throw('Invalid query item').with.property('item', date);
+    const object = {};
+    expect(() => new Composition('Otherwise, throw', object)).to.throw('Invalid query item').with.property('item', object);
+    const array = [];
+    expect(() => new Composition('bad', array)).to.throw('Invalid query item').with.property('item', array);
+  });
+
+  it('should have *set* method', () => {
```

Comment:
Nit: No need to highlighting method names.

Reply:
Asterisks removed